searx/engines/dailymotion.py

```diff
@@ -21,23 +23,73 @@
 # engine dependent config
 categories = ['videos']
 paging = True
+number_of_results = 10
+
+time_range_support = True
+time_delta_dict = {
+    "day":  timedelta(days=1),
+    "week": timedelta(days=7),
+    "month": timedelta(days=31),
+    "year": timedelta(days=365),
+}
 
-# search-url
-# see http://www.dailymotion.com/doc/api/obj-video.html
-search_url = 'https://api.dailymotion.com/videos?fields=created_time,title,description,duration,url,thumbnail_360_url,id&sort=relevance&limit=5&page={pageno}&{query}'  # noqa
-supported_languages_url = 'https://api.dailymotion.com/languages'
+safesearch = True
+safesearch_params = {2: '&is_created_for_kids=true', 1: '&is_created_for_kids=true', 0: ''}
 
+# search-url
+# - https://developers.dailymotion.com/tools/
+# - https://www.dailymotion.com/doc/api/obj-video.html
+
+result_fields = [
+    'allow_embed',
+    'description',
+    'title',
+    'created_time',
+    'duration',
+    'url',
+    'thumbnail_360_url',
+    'id',
+]
+search_url = (
+    'https://api.dailymotion.com/videos?'
+    'fields={fields}&password_protected={password_protected}&private={private}&sort={sort}&limit={limit}'
+).format(
+    fields=','.join(result_fields),
+    password_protected= 'false',
+    private='false',
+    sort='relevance',
+    limit=number_of_results,
+)
+
+# The request query filters by 'languages' & 'country', therefore instead of
+# fetching only languages we need to fetch locales.
+supported_languages_url = 'https://api.dailymotion.com/locales'
 
-# do search-request
 def request(query, params):
-    if params['language'] == 'all':
-        locale = 'en-US'
-    else:
-        locale = match_language(params['language'], supported_languages)
 
-    params['url'] = search_url.format(
-        query=urlencode({'search': query, 'localization': locale}), pageno=params['pageno']
-    )
+    language = params['language']
+    if language == 'all':
+        language = 'en-US'
+    locale = babel.Locale.parse(language, sep='-')
+
+    query_args = {
+        'search': query,
+        'languages': locale.language,
```

Comment:
When the language preference is `ca-ES` (or just `ca`), this value is sent to dailymotion even if dailymotion does not support `ca` = there is no safe guard between the preferences and the value sent to the engine. It works, but it may raise an error in the future. For example a new value in `languages.py` may raise an error in dailymotion.

Reply:
Fix in commit https://github.com/searxng/searxng/commit/f4ed1110f2b1635e9847e36d997a924eb8c0bbca